store/sqlstore/plugin_store.go

```diff
@@ -71,6 +71,43 @@ func (ps SqlPluginStore) SaveOrUpdate(kv *model.PluginKeyValue) store.StoreChann
 	})
 }
 
+func (ps SqlPluginStore) CompareAndUpdate(kv *model.PluginKeyValue, oldValue []byte) store.StoreChannel {
+	return store.Do(func(result *store.StoreResult) {
+		if result.Err = kv.IsValid(); result.Err != nil {
+			return
```

Comment:
This is obviously copy/paste, but I'm curious why we discard the error here. (I've likely forgotten!)

Reply:
Don't think we discard the error, `result.Err` gets assigned as part of the if check `if result.Err = kv.IsValid();`.

Reply:
D'oh, yeah -- misread that :). Another reason why the blocking simplification above will be much easier to follow!